sound/soc/sof/intel/hda-dai.c

```diff
@@ -151,14 +151,15 @@ static int hda_link_dma_params(struct hdac_ext_stream *stream,
 	return 0;
```

Comment:
@kv2019i This is the tricky part to understand amongst all the other changes in this PR. Basically, we set up all the component including the DAI component in the FW hw_params now. When a pcm is started, the order though is BE hw_params followed by FE hw_params. And because of this ordering, we cant do DAI_CONFIG IPC in BE hw_params because the DAI itself doesnt exist just yet. So we save the params in BE hw_params and send it after the DAI component is created in FE hw_params. Make sense?